src/resolve/resolved-dns-scope.c

```diff
@@ -569,6 +569,10 @@ static DnsScopeMatch match_subnet_reverse_lookups(
                 if (a->prefixlen == UCHAR_MAX) /* don't know subnet mask */
                         continue;
 
+                /* Don't send mDNS queries for the IPv4 broadcast address */
+                if (in_addr_equal(f, &a->in_addr_broadcast, &ia))
```

Comment:
event though this will work, I think we should include a family check here, to make clear that this is only relevant for ipv4.

i.e. 
```c
if (f == AF_INET && in_addr_equal(f, &a->in_addr_broadcast, &ia) > 0)
        return DNS_SCOPE_NO;
```

in_addr_equal() can return an error (i.e. negative errno-like value), which is fine to ignore here, but we should so so explicit, hence the `> 0` comparison here.

Reply:
hmm, so if an interface has no broadcast address then this will filter 0.0.0.0. Which is OK, but I'd prefer if maybe we'd filter that beforehand explicitly, so that the ambiguity that this creates (i.e. that 0.0.0.0 is not filtered for networks that have a broadcast address) doesn't happen. Hence, please maybe add an explicit in_addr_is_null() before the loop and refuse that address, so this corner case is never touched.